### DISCUSSION TITLE
Anyone still using this library?

### DISCUSSION THREAD

**Discussion by sagikazarmark** [bernardphp/bernard#413] [General]
Another year passed and I'm trying to set things in order. I stumbled upon Bernard recently as a library that I took over, but never finished and I decided to set things right: either deprecate the library entirely or release a stable version based on the current API (there are really not much things to do). Then put it in maintenance mode or pass it entirely onto the next maintainer.

So the first questions to answer: is there anyone still using this library? Is it worth the effort to release a stable version? Are there any alternative libraries that work better?

If the overwhelming answer is yes (the community needs this library) then I have the following steps to reach a stable version in mind:

- Drop PHP 5.x support, require minimum 7.4 (or go straight to 8.0)
- Add type hints and all the modern stuff
- Replace symfony event dispatcher with PSR-14
- Relocate the rest of the drivers to their own packages
- Upgrade dependencies
- Add some more static analysis tools (use guzzle as an example)
- Upgrade anything else as necessary
- Skip 1.x (there are already beta version and someone might use the library as is on PHP 5.x)

Let me know what you think! Please spread the word amongst the users of these library!

Thanks,
@sagikazarmark

**Comment by rutek** [comment]
@sagikazarmark Library is quite popular as you can see on Packagist. People use version 1 regardless fact that is not marked as final: https://packagist.org/packages/bernard/bernard/stats

I think that if there are no bugs reported on Github, v1 should be marked as stable as current code seems to be battle tested. I use it personally to sometimes push millions of jobs to multiple queues and it works great. One thing I noticed are notices generated by Symfony EventDispatcher when on 7.4.

Changes you proposed seems OK to me but for 2.0 version as it contains multiple breaking changes. I won't recommend skipping PHP 7.4 as it will live for next few years.

**Reply by sagikazarmark** [reply]
Thanks a lot for your feedback @rutek !

Download count is one metric that we can rely on, but not necessarily measures how much people actually rely on bernard which is why I'd love to hear feedback from as much people as possible.

I'm not necessarily inclined on skipping 7.4 unless we want to use some 8.0 features (mixed for instance) in which case I don't mind skipping 7.4. We'll see.

Thanks again for the feedback.

**Comment by vitman** [comment]
Communty definately needs this lib.
I'm working in known game-dev company and we have to make fork in order to keep it more up to date and  use it with more actual symfony components, it would be super nice to get actual lib out of the box instead of forks pain.

**Reply by rutek** [reply]
Same for us but we are using it in Laminas. After 1.x has been dropped in favour of creating 2.x we had to create internal fork which keeps PHP 7.2 compatibility for next few months while containing necessary notice fixes as they created tons of logs.

**Comment by fondberg** [comment]
We are

**Comment by djoos** [comment]
Definitely still in use. It would be great to get an idea of whether this repo will get maintained again, or whether it's time to fork...

Thanks in advance!

**Comment by sagikazarmark** [comment]
I'll see if I can free up some time to get Bernard past 1.0.

Short todo list:

- Moving drivers out
- Moving symfony specific stuff into a symfony-integration package
- Refactoring some interfaces (add type hints, etc)

For the long term, it would be nice to get more maintainers on board.

**Comment by pluc** [comment]
I would like to use it but the current discrepancies between the documentation (which are for v1) and the code (which is 0.13) makes that very confusing. So I'd release the alpha that has spent six years being the latest release if only to harmonize practices.

**Comment by sagikazarmark** [comment]
Please see https://github.com/bernardphp/bernard/discussions/417
